pkg/hubble/exporter/config_watcher.go

```diff
@@ -23,43 +22,40 @@ var reloadInterval = 5 * time.Second
 type configWatcher struct {
 	logger         logrus.FieldLogger
 	configFilePath string
-	callback       func(ctx context.Context, hash uint64, config DynamicExportersConfig)
-	ticker         *time.Ticker
-	stop           chan bool
+	callback       func(hash uint64, config DynamicExportersConfig)
+	stop           chan struct{}
 }
 
 // NewConfigWatcher creates a config watcher instance. Config watcher notifies
 // dynamic exporter when config file changes and dynamic exporter should be
 // reconciled.
 func NewConfigWatcher(
 	configFilePath string,
-	callback func(ctx context.Context, hash uint64, config DynamicExportersConfig),
+	callback func(hash uint64, config DynamicExportersConfig),
 ) *configWatcher {
 	watcher := &configWatcher{
 		logger:         logrus.New().WithField(logfields.LogSubsys, "hubble").WithField("configFilePath", configFilePath),
 		configFilePath: configFilePath,
 		callback:       callback,
+		stop:           make(chan struct{}),
 	}
 
 	// initial configuration load
 	watcher.reload()
+	return watcher
+}
 
+func (c *configWatcher) watch() {
 	// TODO replace ticker reloads with inotify watchers
-	watcher.ticker = time.NewTicker(reloadInterval)
-	watcher.stop = make(chan bool)
```

Comment:
The thing is, it could never be restarted before because we didn't have a start method. I thought by doing it this way, we are more explicit about being single-use. I can always update start so it is compatible and can be restarted if desired.

Reply:
Updated to be context-based, similar to other cells making use of OneShot, which means we don't need to store any state afterall. It also has the advantage of decoupling starting/stoping the watcher from stopping the exporter itself (which is not restartable).